cibuildwheel/linux.py

```diff
@@ -138,6 +138,16 @@ def build(options: BuildOptions) -> None:
                     log.build_start(config.identifier)
 
                     dependency_constraint_flags: List[Union[str, PathLike]] = []
+                    if config.identifier.startswith("pp"):
+                        # Patch PyPy to make sure headers get installed into a venv
+                        patch_version = '_27' if config.version == '2.7' else ''
+                        patch_path = Path(__file__).absolute().parent / 'resources' / f'pypy_venv{patch_version}.patch'
+                        patch_docker_path = PurePath('/pypy_venv.patch')
+                        docker.copy_into(patch_path, patch_docker_path)
+                        try:
+                            docker.call(['patch', '-N', '-p1', '-d', config.path, '-i', patch_docker_path])
```

Comment:
Why `-N`, why `-i`?

> `-N  --forward  Ignore patches that appear to be reversed or already applied.`

I don't see the point of this? We know our patches should apply forward?

> `-i PATCHFILE  --input=PATCHFILE  Read patch from PATCHFILE instead of stdin.`

This was also not the issue in what you showed. The issue seems to have been the leading `a`/`b`?

Better apply one of these, then?

> `  -t  --batch  Ask no questions; skip bad-Prereq patches; assume reversed.`
> `  -f  --force  Like -t, but ignore bad-Prereq patches, and assume unreversed.`

Reply:
The leading a/b character should be fixed by `-p1`. The `-i` (or `--input`) is correct, we want to specify an input patch, even if it works without it. I think `--force` is probably what you want here, though, since you never want it to ask questions, which I think `-N` still could do if it's not one of {unpatched, already patched}.

Reply:
```suggestion
                            docker.call(['patch', '--force', '-p1', '-d', config.path, '--input', patch_docker_path])
```

Perhaps?

Reply:
`-i` is needed to specify the patch file. Patch on Linux assumes that the first file is one to patch. 

I see no difference in behavior between `-N` and `--force`. We have properly formated patch files.